rate_test.go

```diff
@@ -110,6 +110,36 @@ func TestRateLimit(t *testing.T) {
 	}
 }
 
+func TestRateLimitCustomPeriod(t *testing.T) {
+	period := 10 * time.Millisecond
+	rq := throttled.RateQuota{throttled.PerPeriod(3, period), 0}
+	mst, err := memstore.New(27)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	st := testStore{store: mst}
+	rl, err := throttled.NewGCRARateLimiter(&st, rq)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 27; i++ {
+		limited, _, err := rl.RateLimit("bar", 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if i != 0 && i%3 == 0 && !limited {
+			t.Errorf("%d is expected to be limited", i)
+		}
+
+		if limited {
+			time.Sleep(period)
+		}
+	}
+}
```

Comment:
Personally, I would be okay without this test given that it feels like a lot of test to check a division :)

It's fine to leave it for now too, but I might go through and refactor it later.